Make the tool parser ignore the conflict marker lines <<<< ==== >>>>

Discussed in the meeting of 2023-03-07.

Need to either check that the conflicting lines are comments or whitespace, or that they are logically nonconflicting.

We should also ignore conflicts in PropList.txt:
```
java.lang.reflect.InvocationTargetException
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:293)
    at java.lang.Thread.run (Thread.java:833)
Caused by: com.ibm.icu.util.ICUException: Prepended_Concatenation_Mark( from: C:\Users\robin\Projects\Unicode\unicodetools\unicodetools\data\ucd\dev\PropList.txt)
    at org.unicode.props.IndexUnicodeProperties.load (IndexUnicodeProperties.java:446)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:294)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:285)
    at org.unicode.text.UCD.DerivedProperty$13.<init> (DerivedProperty.java:748)
    at org.unicode.text.UCD.DerivedProperty.<init> (DerivedProperty.java:731)
    at org.unicode.text.UCD.DerivedProperty.getCached (DerivedProperty.java:62)
    at org.unicode.text.UCD.DerivedProperty.make (DerivedProperty.java:46)
    at org.unicode.text.UCD.ToolUnicodePropertySource.<init> (ToolUnicodePropertySource.java:899)
    at org.unicode.text.UCD.ToolUnicodePropertySource.make (ToolUnicodePropertySource.java:78)
    at org.unicode.text.UCD.MakeUnicodeFiles.generatePropertyFile (MakeUnicodeFiles.java:1127)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:528)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:454)
    at org.unicode.text.UCD.MakeUnicodeFiles.main (MakeUnicodeFiles.java:56)
    at org.unicode.text.UCD.Main.main (Main.java:183)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:293)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.IllegalArgumentException: line: <<<<<<< HEAD
    at org.unicode.props.UcdLineParser$UcdLine.hasNext (UcdLineParser.java:157)
    at org.unicode.props.PropertyParsingInfo.parsePropertyValueFile (PropertyParsingInfo.java:655)
    at org.unicode.props.PropertyParsingInfo.parseSourceFile (PropertyParsingInfo.java:501)
    at org.unicode.props.IndexUnicodeProperties.load (IndexUnicodeProperties.java:443)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:294)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:285)
    at org.unicode.text.UCD.DerivedProperty$13.<init> (DerivedProperty.java:748)
    at org.unicode.text.UCD.DerivedProperty.<init> (DerivedProperty.java:731)
    at org.unicode.text.UCD.DerivedProperty.getCached (DerivedProperty.java:62)
    at org.unicode.text.UCD.DerivedProperty.make (DerivedProperty.java:46)
    at org.unicode.text.UCD.ToolUnicodePropertySource.<init> (ToolUnicodePropertySource.java:899)
    at org.unicode.text.UCD.ToolUnicodePropertySource.make (ToolUnicodePropertySource.java:78)
    at org.unicode.text.UCD.MakeUnicodeFiles.generatePropertyFile (MakeUnicodeFiles.java:1127)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:528)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:454)
    at org.unicode.text.UCD.MakeUnicodeFiles.main (MakeUnicodeFiles.java:56)
    at org.unicode.text.UCD.Main.main (Main.java:183)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:293)
    at java.lang.Thread.run (Thread.java:833)
Caused by: org.unicode.text.utility.ChainException: bad hex value: ‘<’ at position 0 in "<<<<<<< HEAD"
    at org.unicode.text.utility.Utility.fromHex (Utility.java:486)
    at org.unicode.text.utility.Utility.fromHex (Utility.java:430)
    at org.unicode.text.utility.Utility.fromHex (Utility.java:426)
    at org.unicode.props.UcdLineParser$IntRange.set (UcdLineParser.java:42)
    at org.unicode.props.UcdLineParser$UcdLine.hasNext (UcdLineParser.java:155)
    at org.unicode.props.PropertyParsingInfo.parsePropertyValueFile (PropertyParsingInfo.java:655)
    at org.unicode.props.PropertyParsingInfo.parseSourceFile (PropertyParsingInfo.java:501)
    at org.unicode.props.IndexUnicodeProperties.load (IndexUnicodeProperties.java:443)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:294)
    at org.unicode.props.IndexUnicodeProperties.loadEnum (IndexUnicodeProperties.java:285)
    at org.unicode.text.UCD.DerivedProperty$13.<init> (DerivedProperty.java:748)
    at org.unicode.text.UCD.DerivedProperty.<init> (DerivedProperty.java:731)
    at org.unicode.text.UCD.DerivedProperty.getCached (DerivedProperty.java:62)
    at org.unicode.text.UCD.DerivedProperty.make (DerivedProperty.java:46)
    at org.unicode.text.UCD.ToolUnicodePropertySource.<init> (ToolUnicodePropertySource.java:899)
    at org.unicode.text.UCD.ToolUnicodePropertySource.make (ToolUnicodePropertySource.java:78)
    at org.unicode.text.UCD.MakeUnicodeFiles.generatePropertyFile (MakeUnicodeFiles.java:1127)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:528)
    at org.unicode.text.UCD.MakeUnicodeFiles.generateFile (MakeUnicodeFiles.java:454)
    at org.unicode.text.UCD.MakeUnicodeFiles.main (MakeUnicodeFiles.java:56)
    at org.unicode.text.UCD.Main.main (Main.java:183)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:293)
    at java.lang.Thread.run (Thread.java:833)
```

--- a/unicodetools/src/main/java/org/unicode/props/UcdLineParser.java
+++ b/unicodetools/src/main/java/org/unicode/props/UcdLineParser.java
@@ -111,6 +111,11 @@ public final class UcdLineParser implements Iterable<UcdLineParser.UcdLine> {
                         return false;
                     }
                     line = line2 = rawLines.next();
+                    if (line.startsWith("<<<<<<<")
+                            || line.startsWith("=======")
+                            || line.startsWith(">>>>>>>")) {
+                        line2 = "";
+                    }
                     ++stats.lineCount;
                     final int hashPos = line2.indexOf('#');
                     if (hashPos >= 0) {

